### ISSUE TITLE
Connect error since 3.5.0

### ISSUE THREAD

**Issue by LeonMelis** [php-amqplib/php-amqplib#1065]
**Version(s) affected**: 3.5.0

**Description**  
Since updating to 3.5.0, application with AMQPLib 3.5.0 can no longer connect to RabbitMQ server (v3.11.6)

```
PhpAmqpLib\Exception\AMQPIOException: stream_socket_client(): Unable to connect to ssl://x.x.x.x:5672 (Unknown error) in /opt/project/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php on line 99
```

In the RabbitMQ (v3.11.6) server logs we see following error appear:

```
[error] <0.19823.13> closing AMQP connection <0.19823.13> (x.x.x.x:33722 -> x.x.x.x:5672):
[error] <0.19823.13> {bad_header,<<22,3,1,2,0,1,0,1>>}
```

**How to reproduce**  
We used code that previously worked. Only change was update to 3.5.0

```php
$connection = new AMQPSSLConnection( '127.0.0.1', 5672, 'user', 'password', '/',  null,  [
   'read_write_timeout' => 30
   'connection_timeout' => 30
   'keepalive' => false,
   'heartbeat' => 0
]);
```

Note that we pass `null` to the SSL settings object. I suspect the issue has something to do with the SSL changes in 3.5.0.

**Possible Solution**  
Downgrading to 3.4.0 worked for us.

**Additional context**  
I expect it to have to do with this: https://github.com/php-amqplib/php-amqplib/pull/1060

**Comment by lukebakken** [comment]
It seems odd to use `AMQPSSLConnection` when you're not actually using TLS. Why are you not using one of the non-TLS connection classes?

**Comment by LeonMelis** [comment]
@lukebakken We deploy our product for various environments, some use TLS, some don't, so we made it configurable. The code example I posted was a simplified PoC with hardcoded values.

If TLS is used, we pass port number 5671 and the `$ssl_options` argument to the constructor. 

But if no TLS is not used, we still use `AMQPSSLConnection`, but we pass port 5672 and `null` for the `$ssl_options` argument. 

We have been using it this way for years for both TLS and non-TLS implementations, without problems. And because it worked, I guess I just assumed  `AMQPSSLConnection`  could handle both TLS and non-TLS modes, so it was the logical choice for using it in our code which had to support both.

I guess I just assumed the  `AMQPSSLConnection` class was introduced to php-amqplib when TLS support was added, and `AMQPConnection` was kept for backwards compatibility (constructor arguments). 


If  `AMQPSSLConnection`  can not longer be used for non-TLS connections from 3.5.0 onwards, at least add a big fat warning about this in the release notes, or the readme.

**Comment by lukebakken** [comment]
Thanks for the explanation, that's interesting.

> at least add a big fat warning about this in the release notes, or the readme

It's just an oversight. We'll address it soon in `3.5.1`

**Comment by ramunasd** [comment]
@lukebakken do You really think we should revert anything here?
Personally I think that we did good job and now library works in a way most developers expect. Changes in #1060 was a fix to wrong behaviour, not breaking change.

**Comment by ramunasd** [comment]
Sooner or later people will adopt. That's only a matter of time.

**Comment by ramunasd** [comment]
(facepalm)

**Comment by lukebakken** [comment]
@LeonMelis - https://github.com/php-amqplib/php-amqplib/releases/tag/v3.5.1

Thanks for your patience.

**Comment by LeonMelis** [comment]
Thank you for fixing this so quickly!

Admittedly is was kinda torn between @lukebakken  and @ramunasd, because I do agree with @ramunasd that we did abuse the API by using `AMQPSSLConnection` without TLS. 

However, I support @lukebakken's decision to revert change #1060 in #1066, as there will probably be more implementations using `AMQPSSLConnection` like we did, and you don't want a minor release to break existing implementations. I think it is a good idea to change this with the next major version release (4.0.0), as outlined in #1067

Thank you for building and maintaining this library. It is an essential piece of the PHP ecosystem.
